application/clicommands/JobsCommand.php

```diff
@@ -162,28 +164,32 @@ protected function fetchSchedules(?string $jobName, ?string $scheduleName): arra
         foreach ($jobs as $jobConfig) {
             $cidrs = $this->parseCIDRs($jobConfig->cidrs);
             $ports = $this->parsePorts($jobConfig->ports);
-            $job = (new Job($jobConfig->name, $cidrs, $ports, $snimap))
-                ->setId($jobConfig->id)
-                ->setExcludes($this->parseExcludes($jobConfig->exclude_targets));
 
             /** @var Query $schedules */
             $schedules = $jobConfig->schedule;
             if ($scheduleName) {
                 $schedules->filter(Filter::equal('name', $scheduleName));
             }
 
+            $schedules = $schedules->execute();
+            $hasSchedules = $schedules->hasResult();
+
             /** @var X509Schedule $scheduleModel */
             foreach ($schedules as $scheduleModel) {
-                $schedule = Schedule::fromModel($scheduleModel);
-                $job = (clone $job)
-                    ->setSchedule($schedule)
-                    ->setUuid(Uuid::fromBytes($job->getChecksum()));
+                $job = (new Job($jobConfig->name, $cidrs, $ports, $snimap, Schedule::fromModel($scheduleModel)))
+                    ->setId($jobConfig->id)
+                    ->setExcludes($this->parseExcludes($jobConfig->exclude_targets));
+
+                // The Job class sets the uuid in its constructor, but since the excluded targets are also hashed as
+                // part of the job's uuid and the excluded targets are set after the job construction, we have to
+                // reset the uuid afterwards. Otherwise, it won't notice when updating the "exclude_targets" column.
+                $job->setUuid(Uuid::fromBytes($job->getChecksum()));
```

Comment:
If it's that important, why does `setExcludes` not regenerate the uuid? Of course, that'd apply to all setters changing anything related to the uuid. To avoid regenerating the uuid each time, you could make `Job::getUuid()` not throw but lazily call `Job::getChecksum()`. But that's just an idea, making the code probably cleaner and removing the need for the comment ;)